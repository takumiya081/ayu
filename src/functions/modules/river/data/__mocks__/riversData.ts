import {RiverDataType} from '../riversData';

const mockPrefectures = ['和歌山県', '奈良県', '三重県', '京都府', '滋賀県'];

const mockRiversData: RiverDataType[] = [];
for (let i = 0; i < 100; i++) {
  mockRiversData.push({
    id: `river-test-id-${i}`,
    name: `river name ${i}`,
    lat: i + 120,
    lng: i + 90,
    prefecture: mockPrefectures[i % mockPrefectures.length],
    unionId: `union-test-id-${i}`,
  });
}

export const riversData: ReadonlyArray<RiverDataType> = mockRiversData;
