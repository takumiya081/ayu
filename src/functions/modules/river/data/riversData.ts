export interface RiverDataType {
  id: string;
  name: string;
  lat: number;
  lng: number;
  unionId?: string;
  prefecture: string;
}

export const riversData: ReadonlyArray<RiverDataType> = [
  {
    id: 'river-日置川-33.808927-135.611457',
    prefecture: '和歌山県',
    name: '日置川',
    lat: 33.808927,
    lng: 135.611457,
    unionId: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'river-古座川-33.561905-135.686945',
    prefecture: '和歌山県',
    name: '古座川',
    lat: 33.561905,
    lng: 135.686945,
    unionId: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'river-紀ノ川-34.31021-135.606522',
    prefecture: '和歌山県',
    name: '紀ノ川',
    lat: 34.31021,
    lng: 135.606522,
    unionId: 'union-和歌山県-紀ノ川漁業協同組合',
  },
  {
    id: 'river-有田川-34.04713-135.343151',
    prefecture: '和歌山県',
    name: '有田川',
    lat: 34.04713,
    lng: 135.343151,
    unionId: 'union-和歌山県-有田川漁業協同組合',
  },
  {
    id: 'river-貴志川-34.225495-135.327583',
    prefecture: '和歌山県',
    name: '貴志川',
    lat: 34.225495,
    lng: 135.327583,
    unionId: 'union-和歌山県-貴志川漁業協同組合',
  },
  {
    id: 'river-日高川-33.921087-135.434861',
    prefecture: '和歌山県',
    name: '日高川',
    lat: 33.921087,
    lng: 135.434861,
    unionId: 'union-和歌山県-日高川漁業協同組合',
  },
  {
    id: 'river-富田川-33.787221-135.512022',
    prefecture: '和歌山県',
    name: '富田川',
    lat: 33.787221,
    lng: 135.512022,
    unionId: 'union-和歌山県-富田川漁業協同組合',
  },
  {
    id: 'river-熊野川-33.867404-135.76179',
    prefecture: '和歌山県',
    name: '熊野川',
    lat: 33.867404,
    lng: 135.76179,
    unionId: 'union-和歌山県-熊野川漁業協同組合・熊野川漁業協同組合連合協議会',
  },
  {
    id: 'river-北山川-33.890741-135.895256',
    prefecture: '和歌山県',
    name: '北山川',
    lat: 33.890741,
    lng: 135.895256,
    unionId: 'union-和歌山県-熊野川漁業協同組合・熊野川漁業協同組合連合協議会',
  },
  {
    id: 'river-美山川-35.320526-135.67847',
    prefecture: '京都府',
    name: '美山川',
    lat: 35.320526,
    lng: 135.67847,
    unionId: 'union-京都府-美山漁業協同組合',
  },
];
