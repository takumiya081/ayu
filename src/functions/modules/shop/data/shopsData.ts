export interface ShopDataType {
  id: string;
  name: string;
  prefecture: string;
  address: string;
  lat: number;
  lng: number;
  link?: string;
  unionIds: string;
}

export const unionIdsDelimiter = ',';

export const shopsData: ReadonlyArray<ShopDataType> = [
  {
    id: 'shop-民宿　日吉屋-33.624644-135.482553',
    prefecture: '和歌山県',
    name: '民宿　日吉屋',
    address: '和歌山県西牟婁郡白浜町中嶋119',
    lat: 33.624644,
    lng: 135.482553,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　旭屋-33.637484-135.469762',
    prefecture: '和歌山県',
    name: '民宿　旭屋',
    address: '和歌山県西牟婁郡白浜町久木317',
    lat: 33.637484,
    lng: 135.469762,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　前田屋-33.636223-135.467362',
    prefecture: '和歌山県',
    name: '民宿　前田屋',
    address: '和歌山県西牟婁郡白浜町久木869-33',
    lat: 33.636223,
    lng: 135.467362,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　うつぎ-33.650694-135.499567',
    prefecture: '和歌山県',
    name: '民宿　うつぎ',
    address: '和歌山県西牟婁郡白浜町宇津木39-1',
    lat: 33.650694,
    lng: 135.499567,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　やすだ-33.647522-135.505203',
    prefecture: '和歌山県',
    name: '民宿　やすだ',
    address: '和歌山県西牟婁郡白浜町大440',
    lat: 33.647522,
    lng: 135.505203,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　森のお宿-33.647703-135.505411',
    prefecture: '和歌山県',
    name: '民宿　森のお宿',
    address: '和歌山県西牟婁郡白浜町大880-2',
    lat: 33.647703,
    lng: 135.505411,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-土井-33.657593-135.534039',
    prefecture: '和歌山県',
    name: '土井',
    address: '和歌山県西牟婁郡白浜町市鹿野1097',
    lat: 33.657593,
    lng: 135.534039,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　岡本-33.6713-135.542733',
    prefecture: '和歌山県',
    name: '民宿　岡本',
    address: '和歌山県西牟婁郡白浜町市鹿野1738-3',
    lat: 33.6713,
    lng: 135.542733,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　殿山-33.677316-135.546105',
    prefecture: '和歌山県',
    name: '民宿　殿山',
    address: '和歌山県西牟婁郡白浜町市鹿野1819',
    lat: 33.677316,
    lng: 135.546105,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-森山商店-33.684138-135.575985',
    prefecture: '和歌山県',
    name: '森山商店',
    address: '和歌山県田辺市合川481',
    lat: 33.684138,
    lng: 135.575985,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-谷口-33.68887-135.568116',
    prefecture: '和歌山県',
    name: '谷口',
    address: '和歌山県田辺市合川637',
    lat: 33.68887,
    lng: 135.568116,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-橋本商店-33.691323-135.5954',
    prefecture: '和歌山県',
    name: '橋本商店',
    address: '和歌山県田辺市面川118',
    lat: 33.691323,
    lng: 135.5954,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　やまびこ-33.737434-135.57554',
    prefecture: '和歌山県',
    name: '民宿　やまびこ',
    address: '和歌山県田辺市下川下1398-1',
    lat: 33.737434,
    lng: 135.57554,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-峯取扱店-33.73221451-135.57326783',
    prefecture: '和歌山県',
    name: '峯取扱店',
    address: '和歌山県田辺市下川下1250',
    lat: 33.73221451,
    lng: 135.57326783,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿　花しょうぶ-33.73590019-135.57789232',
    prefecture: '和歌山県',
    name: '民宿　花しょうぶ',
    address: '和歌山県田辺市下川下914-3',
    lat: 33.73590019,
    lng: 135.57789232,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-古道の杜あんちゃん-33.77542247-135.50391674',
    prefecture: '和歌山県',
    name: '古道の杜あんちゃん',
    address: '和歌山県田辺市中辺路町栗栖川1226-2',
    lat: 33.77542247,
    lng: 135.50391674,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-お宿 まんまる-33.82193162-135.60521968',
    prefecture: '和歌山県',
    name: 'お宿 まんまる',
    address: '和歌山県田辺市中辺路町近露2722-2',
    lat: 33.82193162,
    lng: 135.60521968,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿 ちかつゆ-33.81682114-135.60581134',
    prefecture: '和歌山県',
    name: '民宿 ちかつゆ',
    address: '和歌山県田辺市中辺路町近露401-12',
    lat: 33.81682114,
    lng: 135.60581134,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-民宿 中野-33.80724463-135.60838057',
    prefecture: '和歌山県',
    name: '民宿 中野',
    address: '和歌山県田辺市中辺路町近露2140',
    lat: 33.80724463,
    lng: 135.60838057,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-アイリスパーク-33.817577-135.615788',
    prefecture: '和歌山県',
    name: 'アイリスパーク',
    address: '和歌山県田辺市中辺路町近露24-1',
    lat: 33.817577,
    lng: 135.615788,
    link: '',
    unionIds: 'union-和歌山県-日置川漁業協同組合',
  },
  {
    id: 'shop-古座川水産-135.78135469-33.53561547',
    prefecture: '和歌山県',
    name: '古座川水産',
    address: '和歌山県東牟婁郡古座川町高瀬202',
    lat: 135.78135469,
    lng: 33.53561547,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-福池紀生-33.54136992-135.76435885',
    prefecture: '和歌山県',
    name: '福池紀生',
    address: '和歌山県東牟婁郡古座川町明神153',
    lat: 33.54136992,
    lng: 135.76435885,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-藤田商店-33.53652872-135.75508706',
    prefecture: '和歌山県',
    name: '藤田商店',
    address: '和歌山県東牟婁郡古座川町一雨420-2',
    lat: 33.53652872,
    lng: 135.75508706,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-硲-33.548257-135.72727',
    prefecture: '和歌山県',
    name: '硲',
    address: '和歌山県東牟婁郡古座川町相瀬144',
    lat: 33.548257,
    lng: 135.72727,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-大石-33.54683247-135.72703114',
    prefecture: '和歌山県',
    name: '大石',
    address: '和歌山県東牟婁郡古座川町相瀬197',
    lat: 33.54683247,
    lng: 135.72703114,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-田上-33.55586455-135.71615431',
    prefecture: '和歌山県',
    name: '田上',
    address: '和歌山県東牟婁郡古座川町洞尾224',
    lat: 33.55586455,
    lng: 135.71615431,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-民宿 沖-33.55674182-135.68569021',
    prefecture: '和歌山県',
    name: '民宿 沖',
    address: '和歌山県東牟婁郡古座川町三尾川981-1',
    lat: 33.55674182,
    lng: 135.68569021,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-浅利-33.58695561-135.6884195',
    prefecture: '和歌山県',
    name: '浅利',
    address: '和歌山県東牟婁郡古座川町長追518',
    lat: 33.58695561,
    lng: 135.6884195,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-民宿 やまびこ-33.60364959-135.76698001',
    prefecture: '和歌山県',
    name: '民宿 やまびこ',
    address: '和歌山県東牟婁郡古座川町小川809-1',
    lat: 33.60364959,
    lng: 135.76698001,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-坂本-33.58959953-135.62498862',
    prefecture: '和歌山県',
    name: '坂本',
    address: '和歌山県西牟婁郡すさみ町佐本中217-1',
    lat: 33.58959953,
    lng: 135.62498862,
    link: '',
    unionIds: 'union-和歌山県-古座川漁業協同組合',
  },
  {
    id: 'shop-喫茶TOMOTOMO-34.275917-135.477716',
    prefecture: '和歌山県',
    name: '喫茶TOMOTOMO',
    address: '和歌山県伊都郡かつらぎ町東渋田396',
    lat: 34.275917,
    lng: 135.477716,
    link: '',
    unionIds: 'union-和歌山県-紀ノ川漁業協同組合',
  },
  {
    id: 'shop-三幸ストアー（無人販売）-34.31270291-135.61459505',
    prefecture: '和歌山県',
    name: '三幸ストアー（無人販売）',
    address: '和歌山県橋本市向副1009',
    lat: 34.31270291,
    lng: 135.61459505,
    link: '',
    unionIds: 'union-和歌山県-紀ノ川漁業協同組合',
  },
  {
    id: 'shop-玉川峡ドライブイン-34.27626444-135.5685166',
    prefecture: '和歌山県',
    name: '玉川峡ドライブイン',
    address: '和歌山県伊都郡九度山町九度山766',
    lat: 34.27626444,
    lng: 135.5685166,
    link: '',
    unionIds: 'union-和歌山県-紀ノ川漁業協同組合',
  },
  {
    id: 'shop-枕川楼-35.31357-135.640757',
    prefecture: '京都府',
    name: '枕川楼',
    address: '京都府南丹市美山町中上前26',
    lat: 35.31357,
    lng: 135.640757,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
  {
    id: 'shop-まるや-35.313178-135.639976',
    prefecture: '京都府',
    name: 'まるや',
    address: '京都府南丹市美山町中上前36',
    lat: 35.313178,
    lng: 135.639976,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
  {
    id: 'shop-ひまわり-35.310767-135.63656',
    prefecture: '京都府',
    name: 'ひまわり',
    address: '京都府南丹市美山町中下前上ミ10-1',
    lat: 35.310767,
    lng: 135.63656,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
  {
    id: 'shop-リバーサイド-35.291173-135.601152',
    prefecture: '京都府',
    name: 'リバーサイド',
    address: '京都府南丹市美山町荒倉嶋台55-1',
    lat: 35.291173,
    lng: 135.601152,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
  {
    id: 'shop-平井商店-35.275252-135.57984',
    prefecture: '京都府',
    name: '平井商店',
    address: '京都府南丹市美山町上平屋森本27',
    lat: 35.275252,
    lng: 135.57984,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
  {
    id: 'shop-おとり鮎里 岸本薫-35.263698-135.603861',
    prefecture: '京都府',
    name: 'おとり鮎里 岸本薫',
    address: '京都府南丹市美山町深見中ノ町20',
    lat: 35.263698,
    lng: 135.603861,
    link: '',
    unionIds: 'union-京都府-美山漁業協同組合',
  },
];
