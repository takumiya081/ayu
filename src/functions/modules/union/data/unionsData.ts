export interface UnionDataType {
  id: string;
  name: string;
  prefecture: string;
  link?: string;
  terms?: string;
  phone?: string;
}

export const unionsData: ReadonlyArray<UnionDataType> = [
  {
    id: 'union-和歌山県-日置川漁業協同組合',
    prefecture: '和歌山県',
    name: '日置川漁業協同組合',
    link: 'http://hikigawagyokyou.eek.jp/',
    terms: '5月26日 - 12月31日',
    phone: '0739-53-0023',
  },
  {
    id: 'union-和歌山県-古座川漁業協同組合',
    prefecture: '和歌山県',
    name: '古座川漁業協同組合',
    link: 'https://www.ayuayu.server-shared.com/',
    terms: '6月14日 - 12月31日',
    phone: '0735-72-3800',
  },
  {
    id: 'union-和歌山県-紀ノ川漁業協同組合',
    prefecture: '和歌山県',
    name: '紀ノ川漁業協同組合',
    link: 'http://www.naxnet.or.jp/~kinogyo/',
    terms: '6月1日 - 12月31日',
    phone: '0736-66-9111',
  },
  {
    id: 'union-和歌山県-有田川漁業協同組合',
    prefecture: '和歌山県',
    name: '有田川漁業協同組合',
    link: 'http://aridagawa.com/index.html',
    terms: '5月20日 - 12月31日',
    phone: '0737-52-4863',
  },
  {
    id: 'union-和歌山県-貴志川漁業協同組合',
    prefecture: '和歌山県',
    name: '貴志川漁業協同組合',
    link: '',
    terms: '6月7日 - 12月31日',
    phone: '073-495-2114',
  },
  {
    id: 'union-和歌山県-日高川漁業協同組合',
    prefecture: '和歌山県',
    name: '日高川漁業協同組合',
    link: 'http://www.zb.ztv.ne.jp/hidagyo/',
    terms: '5月20日 - 12月31日',
    phone: '0738-52-0224',
  },
  {
    id: 'union-和歌山県-富田川漁業協同組合',
    prefecture: '和歌山県',
    name: '富田川漁業協同組合',
    link: '',
    terms: '5月31日 - 12月31日',
    phone: '0739-47-0710',
  },
  {
    id: 'union-和歌山県-熊野川漁業協同組合・熊野川漁業協同組合連合協議会',
    prefecture: '和歌山県',
    name: '熊野川漁業協同組合・熊野川漁業協同組合連合協議会',
    link: '',
    terms: '6月1日 - 12月31日',
    phone: '0735-21-4193',
  },
  {
    id: 'union-京都府-美山漁業協同組合',
    prefecture: '京都府',
    name: '美山漁業協同組合',
    link: 'http://www.miyama-gyokyou.jp/',
    terms: '6月6日 - 12月31日',
    phone: '0771-75-0210',
  },
];
