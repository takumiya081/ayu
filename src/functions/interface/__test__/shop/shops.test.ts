import {riversData} from '@/functions/modules/river/data/riversData';
import {
  createApolloTestClient,
  TestShopsByLocationQuery,
  TestShopsByRiverIdQuery,
} from '@/functions/testUtils';

jest.mock('@/functions/modules/shop/data/shopsData');
jest.mock('@/functions/modules/union/data/unionsData');
jest.mock('@/functions/modules/river/data/riversData');

describe('shops query', () => {
  test('riverIdでクエリ', async () => {
    const {query} = createApolloTestClient();
    const result = await query({
      query: TestShopsByRiverIdQuery,
      variables: {riverId: riversData[1].id},
    });
    expect(result).toMatchSnapshot();
  });

  test('locationでクエリ', async () => {
    const {query} = createApolloTestClient();
    const result = await query({
      query: TestShopsByLocationQuery,
      variables: {location: {lat: 35, lng: 137}},
    });
    expect(result).toMatchSnapshot();
  });
});
