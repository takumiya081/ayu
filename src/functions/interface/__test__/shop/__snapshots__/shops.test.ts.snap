// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shops query locationでクエリ 1`] = `
Object {
  "data": Object {
    "shops": Array [
      Object {
        "id": "shop-test-id-39",
        "link": "https://example.com?i=39",
        "location": Object {
          "lat": 34.95,
          "lng": 136.95,
        },
        "name": "shop name 39",
      },
      Object {
        "id": "shop-test-id-40",
        "link": "https://example.com?i=40",
        "location": Object {
          "lat": 35,
          "lng": 137,
        },
        "name": "shop name 40",
      },
      Object {
        "id": "shop-test-id-41",
        "link": "https://example.com?i=41",
        "location": Object {
          "lat": 35.05,
          "lng": 137.05,
        },
        "name": "shop name 41",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`shops query riverIdでクエリ 1`] = `
Object {
  "data": Object {
    "shops": Array [
      Object {
        "id": "shop-test-id-0",
        "link": "https://example.com?i=0",
        "location": Object {
          "lat": 33,
          "lng": 135,
        },
        "name": "shop name 0",
      },
      Object {
        "id": "shop-test-id-1",
        "link": "https://example.com?i=1",
        "location": Object {
          "lat": 33.05,
          "lng": 135.05,
        },
        "name": "shop name 1",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
